main/webapp/modules/core/scripts/index/default-importing-sources/sources.js

```diff
@@ -91,7 +91,17 @@ UrlImportingSourceUI.prototype.attachUI = function(bodyDiv) {
 
   this._elmts.form.submit(function(evt){
     evt.preventDefault();
-    if(!isUrlValid(self._elmts.urlInput[0].value)) {
+    var importUrl = self._elmts.urlInput[0].value;
+    var validUrl =  False;
+    
+    if(isUrlValid(importUrl)) {
+      validUrl = true;
+    } else {
+      importUrl = self._elmts.urlInput[0].value.trim();
+      self._elmts.urlInput[0].value = importUrl;
+    }  
```

Comment:
I think this code could be a lot simpler. You can trim the value no matter if it is valid initially or not - trimming whitespace around a URL will never hurt.